iceoryx_posh/source/popo/building_blocks/chunk_queue_pusher.cpp

```diff
@@ -59,14 +59,11 @@ cxx::expected<ChunkQueueError> ChunkQueuePusher::push(mepoo::SharedChunk chunk)
             auto returnedChunk = mepoo::SharedChunk(chunkManagement);
         }
 
-
-        if (getMembers()->m_semaphoreAttached.load(std::memory_order_acquire) && getMembers()->m_semaphore)
+        if (getMembers()->m_conditionVariableAttached.load(std::memory_order_acquire) && getMembers()->m_conditionVariableDataPtr)
```

Comment:
You also would not need this double check if your single source is the `m_conditionVariableDataPtr`. This double check would also be a race without a lock . You have to do the lock before the check and not after the check as indicated by the comment below

Reply:
Jup, that's easier. Done.